src/Metadata/Resource/ResourceNameCollection.php

```diff
@@ -19,13 +19,15 @@
 final class ResourceNameCollection implements \IteratorAggregate, \Countable
 {
     private $classes;
+    private $context;
 
     /**
      * @param string[] $classes
      */
-    public function __construct(array $classes = [])
+    public function __construct(array $classes = [], array $context = [])
```

Comment:
What do you mean parsing configuration twice? I had a quick glance through the various calls of `getContext`, but haven't been able to figure out what you're trying to do...

If repeated creation of all the metadata on each call is what you're worrying about, we already have PSR-6 caching.